provider/provider.go

```diff
@@ -30,10 +30,10 @@ func Provider() p.Provider {
 	// In this case, a single resource and component
 	return infer.Provider(infer.Options{
```

Comment:
I wonder if it's worthwhile updating the codebase to use the builder we introduced as well?

Reply:
Yes, probably worth it, let's do that in a separate PR though